lib/rules.js

```diff
@@ -197,9 +197,11 @@ module.exports = [
           style = styleForToken;
         }
         if ((styleForToken !== style) &&
-            !((style === "setext_with_atx") &&
+            !((style === "setext_with_atx" || style === "setext_with_atx_closed") &&
               (/h[12]/.test(token.tag) && (styleForToken === "setext")) ||
-               (/h[^12]/.test(token.tag) && (styleForToken === "atx")))) {
+              (/h[^12]/.test(token.tag) &&
+                  (styleForToken === "atx" || styleForToken === "atx_closed")))
```

Comment:
Nitpick: It looks like line 202 lost a space of indentation (as rendered in the PR on GitHub)?
My convention is usually to go in one space for each level of nesting - so 203 would go 2 spaces to the left and 202 1 to the right if I'm reading this correctly.